### ISSUE TITLE
Loss Logvar Scaling

### ISSUE THREAD

**Issue by jtawade** [CompVis/stable-diffusion#380]
During training why is the loss scaled by a log variance value that is learned by the model? 

https://github.com/CompVis/stable-diffusion/blob/ce05de28194041e030ccfc70c635fe3707cdfc30/ldm/models/diffusion/ddpm.py#L1031

This log variance value isn't seemed to be used anywhere else except to scale the loss.

**Comment by jmyoon1** [comment]
I'm curious about this, too. Have you found any reason? logvar does simply seems to reduce over training steps if trained.
I see that this affects to the decoder variance, but this does not seem to converge.

**Comment by lizhiqi49** [comment]
Curious about this, too.

**Comment by bhosalems** [comment]
Same question. I think is this just for scaling? I don't think it is actually learning denoising variance as guidance for that should come form posterior variance by actually minimizing the vlb terms.

Something like

```

def _vb_terms_bpd(
        self, model, x_start, x_t, t, clip_denoised=True, return_log_variance=False, model_kwargs=None
    ):
        """
        Get a term for the variational lower-bound.

        The resulting units are bits (rather than nats, as one might expect).
        This allows for comparison to other papers.

        :return: a dict with the following keys:
                 - 'output': a shape [N] tensor of NLLs or KLs.
                 - 'pred_xstart': the x_0 predictions.
        """
        true_mean, _, true_log_variance_clipped = self.q_posterior_mean_variance(
            x_start=x_start, x_t=x_t, t=t
        )
        out = self.p_mean_variance(
            model, x_t, t, clip_denoised=clip_denoised, model_kwargs=model_kwargs
        )
        kl = normal_kl(
            true_mean, true_log_variance_clipped, out["mean"], out["log_variance"]
        )
        kl = mean_flat(kl) / np.log(2.0)

        decoder_nll = -discretized_gaussian_log_likelihood(
            x_start, means=out["mean"], log_scales=0.5 * out["log_variance"]
        )
        assert decoder_nll.shape == x_start.shape
        decoder_nll = mean_flat(decoder_nll) / np.log(2.0)

        # At the first timestep return the decoder NLL,
        # otherwise return KL(q(x_{t-1}|x_t,x_0) || p(x_{t-1}|x_t))
        output = th.where((t == 0), decoder_nll, kl)
        if return_log_variance:
            return {"output": output, "pred_xstart": out["pred_xstart"], "log_variance": out["log_variance"]}
        return {"output": output, "pred_xstart": out["pred_xstart"]}


```
